proto/buf/alpha/registry/v1alpha1/schema.proto

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/descriptor.proto";
+
+// Provides endpoints for downloading dynamic schemas and for using schemas
+// in validation and data transformation functions.
+service SchemaService {
+  // GetSchema allows the caller to download a schema for one or more requested
+  // types, RPC services, or RPC methods.
+  rpc GetSchema(GetSchemaRequest) returns (GetSchemaResponse);
+
+  // Reformat allows the caller to indicate a type schema which will be used to
+  // transform given data to a different format.
+  rpc Reformat(ReformatRequest) returns (ReformatResponse);
+}
+
+// Module identifies a particular module in the registry.
+message Module {
+  // A user or organization. Required.
+  string owner = 1;
+  // A repository or module name. Required.
+  string repository = 2;
+
+  // A reference to a version. Optional. If present, it may refer to either a
+  // tag or a commit.
+  //
+  // If absent, the latest version for the module identified by the above
+  // owner and repository fields is assumed.
+  string reference = 3;
+}
+
+message GetSchemaRequest {
+  // The module from which the schema is retrieved. Required.
+  Module module = 1;
```

Comment:
No other part of the API refers to a remote reference like this, the current standard for apis that need to fetch info related to a ref is that all three are individual toplevel fields: https://github.com/bufbuild/buf/blob/4b1c53e9b38ee78dbacbf3b1dd3810eab7f0de95/proto/buf/alpha/registry/v1alpha1/image.proto#L47-L49

There is a type like this in module.proto, but 

https://github.com/bufbuild/buf/blob/014e59c1d9aa7e6f5a9f8c0d8451d7d1d5283d32/proto/buf/alpha/module/v1alpha1/module.proto#L58-L65

why the api does not use it for requests is a decision made before me that I don't have good context on. I'm not sure we should break that convention in the `registry.v1alpha1` package.

Reply:
I saw stuff in `resolve.proto` make use of `LocalModuleReference`, so I've changed to re-use that type. Does that seem better?